charts/flyteagent/templates/agent/deployment.yaml

```diff
@@ -26,28 +26,44 @@ spec:
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: "{{ .Values.image.pullPolicy }}"
         name: flyteagent
-        volumeMounts: {{- include "agentSecret.volumeMount" . | nindent 8 }}
         ports:
         - containerPort: {{ .Values.ports.containerPort }}
           name: {{ .Values.ports.name }}
         resources: {{- toYaml .Values.resources | nindent 10 }}
+        env:
+        - name: FLYTE_AWS_ENDPOINT
+          valueFrom:
+            secretKeyRef:
```

Comment:
It's not necessary to use `secretKeyRef`, just use `value` here.

Reply:
thanks!